docs/source/index.rst

```diff
@@ -0,0 +1,291 @@
+
+    {% extends 'base.html' %}
+
+    {% block content %}
+        <h1>{{ plugin.title }}</h1>
+        <div>{{ plugin.description }}</div>
+    {% endblock %}
+
+Notice that the template extends a base.html. We are going to use the application's base template.
+
+Let's create a blueprint in ``example_plugin.py``::
+
+    plugin_blueprint = PluginBlueprint('example', __name__)
+
+In order to create a blueprint of a plugin we need to create a ``PluginBlueprint`` instance.
+
+Then we can create a function rendering the template::
+
+    @plugin_blueprint.route('/plugin')
+    def hello_plugin():
+    return render_template('example_plugin:index.html', plugin=current_plugin)
+
+We can render any plugin template inside our application specifying the name of the plugin before the template name in ``render_template`` function, like above.
+We are also passing a current_plugin object to the template.
+
+
+Now, in the plugin class we can create a method returning the plugin's blueprint::
+
+    def get_blueprint(self):
+        return plugin_blueprint
+
+So our plugin code should look like this::
+
+    from flask import render_template
+    from flask_pluginengine import Plugin, PluginBlueprint, current_plugin
+
+    plugin_blueprint = PluginBlueprint('example', __name__)
+
+
+    @plugin_blueprint.route('/plugin')
+    def hello_plugin():
+        return render_template('example_plugin:index.html', plugin=current_plugin)
+
+
+    class ExamplePlugin(Plugin):
+        """ExamplePlugin
+
+        Just an example plugin
+        """
+
+        def get_blueprint(self):
+            return plugin_blueprint
+
+
+To get more information see the `Plugin`_ API reference.
+
+Minimal application setup
+-------------------------
+
+To create a Flask application using Flask-PluginEngine we need to define it as PluginFlask application::
+
+    from flask_pluginengine import PluginFlask
+    app = PluginFlask(__name__)
+
+    @app.route("/")
+    def hello():
+        return "Hello world!"
+
+    if __name__ == "__main__":
+        app.run()
+
+Now we need to set up the configuration, specifically the namespace of the plugins and the plugins we want to use.
+Let's use the ``example_plugin`` we created before. For example::
+
+    app.config['PLUGINENGINE_NAMESPACE'] = 'example'
+    app.config['PLUGINENGINE_PLUGINS'] = ['example_plugin']
+
+Next we need to create an instance of PluginEngine for our app and load the plugins::
+
+    plugin_engine = PluginEngine(app=app)
+    plugin_engine.load_plugins(app)
+
+Then, we can access the loaded plugins by calling the :func:`get_active_plugins` method, which will return a dictionary containing the active plugins.
+To check if all of the plugins were loaded correctly we can also call the :func:`get_failed_plugins` method that will return a dictionary with plugins that failed to load::
+
+    active_plugins = plugin_engine.get_active_plugins(app=app).values()
+
+
+Example application functionality
```

Comment:
When I started reading this section I didn't know where to write the fragments of code explained and I struggled a bit until I check the example folder. In my opinion would be useful here to either explain that the plugin will be in a separated folder and the app itself in a different one or show again a directory tree pointing where the code in this section should be written in.